File: examples/send_message.py
import requests
import json

def send_whatsapp_message(phone_number: str, message: str) -> dict:
    """
    Send a WhatsApp message using the API.
    
    Args:
        phone_number (str): The recipient's phone number (format: country code + number, e.g., "6281234567890")
        message (str): The message to send
    
    Returns:
        dict: The API response
    """
    url = "http://localhost:3000/api/messages/send"
    
    # Prepare the request payload
    payload = {
        "to": phone_number,
        "message": message
    }
    
    # Set headers
    headers = {
        "Content-Type": "application/json"
    }
    
    try:
        # Send POST request to the API
        response = requests.post(url, json=payload, headers=headers)
        
        # Raise an exception for bad status codes
        response.raise_for_status()
        
        # Return the JSON response
        return response.json()
    
    except requests.exceptions.RequestException as e:
        print(f"Error sending message: {e}")
        return {"success": False, "message": str(e)}

if __name__ == "__main__":
    # Example usage
    phone_number = "6281234567890"  # Replace with the recipient's phone number
    message = "Hello from Python! This is a test message."
    
    print("Sending WhatsApp message...")
    result = send_whatsapp_message(phone_number, message)
    
    if result.get("success"):
        print("Message sent successfully!")
    else:
        print(f"Failed to send message: {result.get('message')}")

File: examples/send_message.js
const axios = require('axios');

/**
 * Send a WhatsApp message using the API
 * @param {string} phoneNumber - The recipient's phone number (format: country code + number, e.g., "6281234567890")
 * @param {string} message - The message to send
 * @returns {Promise<object>} The API response
 */
async function sendWhatsAppMessage(phoneNumber, message) {
    const url = 'http://localhost:3000/api/messages/send';
    
    // Prepare the request payload
    const payload = {
        to: phoneNumber,
        message: message
    };
    
    try {
        // Send POST request to the API
        const response = await axios.post(url, payload, {
            headers: {
                'Content-Type': 'application/json'
            }
        });
        
        return response.data;
    } catch (error) {
        console.error('Error sending message:', error.response?.data || error.message);
        throw error;
    }
}

// Example usage
async function main() {
    const phoneNumber = '6281234567890'; // Replace with the recipient's phone number
    const message = 'Hello from Node.js! This is a test message.';
    
    try {
        console.log('Sending WhatsApp message...');
        const result = await sendWhatsAppMessage(phoneNumber, message);
        
        if (result.success) {
            console.log('Message sent successfully!');
        } else {
            console.log('Failed to send message:', result.message);
        }
    } catch (error) {
        console.error('Error:', error.message);
    }
}

// Run the example if this file is executed directly
if (require.main === module) {
    main().catch(console.error);
}

// Export the function for use in other files
module.exports = sendWhatsAppMessage;
